src/services/hubservice.bal

```diff
@@ -104,6 +106,38 @@ public type HubServiceImpl object {
         }
     }
 
+
+    public function getFailedContent(string subscriberSignature,string topic , string callback ,string timestamp ,int messagecount) returns @untainted repository:FailedContentPullRespModel[]|error{
+               repository:SubscriptionExtendedDetails subscriptionExtendedDetails = self.subsOperations.getSubscription(topic,callback);
+               string hmacSubsSignature=utils:hmacSha256(topic+callback+timestamp,subscriptionExtendedDetails.secret);
+               if(hmacSubsSignature!=subscriberSignature){
+                   return error("SIGNATUREMATCHERROR", message = "hmac didnot match");
+               }
+               string[] msgIDs=self.deliveryReportPersistence.getFailedDeliveryBySubID(subscriptionExtendedDetails.id,timestamp,messagecount);
```

Comment:
As discussed lets do a update and select in a single SQL query and use that to optimize the DB calls

Reply:
This is available in the latest code